File: Algorithms/209. Minimum Size Subarray Sum/javascript/solution.two_pointer.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function (target, nums) {
    const n = nums.length;
    let ans = Number.MAX_SAFE_INTEGER;
    let left = 0, total = 0;
    for (let i = 0; i < n; i++) {
        total += nums[i];
        while (total >= target) {
            ans = Math.min(ans, i - left + 1);
            total -= nums[left++];
        }
    }
    return ans === Number.MAX_SAFE_INTEGER ? 0 : ans;
};

File: Algorithms/209. Minimum Size Subarray Sum/python/solution.two_pointer.py
import sys


class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        n = len(nums)
        ans = sys.maxsize
        left = 0
        total = 0
        for i in range(n):
            total += nums[i]
            while total >= target:
                ans = min(ans, i - left + 1)
                total -= nums[left]
                left += 1
        return ans if ans != sys.maxsize else 0
